### DISCUSSION TITLE
vwc-calendar

### DISCUSSION THREAD

**Discussion by yinonov** [Vonage/vivid#801] [New Developments]
discuss calendar API and UI/UX
calendar is @ _alpha_ state and is awaiting feedback to design its spec

can be view at -
https://vivid.vonage.com/?path=/story/alpha-components-calendar--basic

![image](https://user-images.githubusercontent.com/15140652/116234819-bed38200-a765-11eb-8d46-b71877a08a7f.png)

**Comment by levkaganv** [comment]
Hi, We've played a little with this component and those are the requirements that we need (for now :))

- [x]  Change the dates titles on each of the columns -  We need to be able to display different weeks.
- [x]  Show rows for 24 hours - Currently it shows only between 7am to 10 pm. We need to be able to display events for all hours of the day
- [x] Adding formatted text in the event rectangle - Need it to add event names (and maybe some more info)
- [ ] Scale vertically to take max possible height - This is happening already happening horizontally.
- [ ] In case the viewport of calendar can't show all of the hour rows the calendar should have a scroll bar but only for the rows. The column titles should always be visible.

* If you need me to elaborate more on the requirements or  to get some kind of visual design in Figma please let me know.

**Reply by yinonov** [reply]
#807 addresses datetime (of week) API support

**Reply by RoyMenczerVonage** [reply]
Hi can you provide an example for adding formatted text?
couldn't get it to work (using vivid version 2.7.0) and did not find an example in the documentation, only an example with a label

**Comment by yinonov** [comment]
calendar displays internalized days of week by `en-US `.
future request might demand other locales.
![image](https://user-images.githubusercontent.com/15140652/116655166-45b97200-a993-11eb-8f04-7ec5fad5673f.png)

It's possible to

1. expose a local attribute to override the language
2. a property accepting array of manual days literal

any special preference?

**Reply by levkaganv** [reply]
In option one we would get a property for localization that we would be able to change ?
If yes then this is what we prefer.

**Comment by levkaganv** [comment]
@yinonov About the points that you checked. Does that mean they already merged ? If yes how can I get the latest version of calendar so I could play with it a little more ?

**Reply by yinonov** [reply]
Checked items above are already merged with recent version

**Comment by jshenkman** [comment]
Hey, 
The formated text for the event should be by [this design ](https://www.figma.com/file/r8OYLPmRmDqfLKt6KHdJDB/Vonage-UI-Kit-2.0?node-id=4650%3A19)
And added the [Calendar component design](https://www.figma.com/file/r8OYLPmRmDqfLKt6KHdJDB/Vonage-UI-Kit-2.0?node-id=4239%3A29304) to our library

**Reply by yinonov** [reply]
Thanks @jshenkman ill follow

**Reply by RoyMenczerVonage** [reply]
@jshenkman @yinonov 
Following A meeting between my team, product, and the UX designer for the "VR scheduling" feature - turns out that there is one more capability that we need for the formatted text in vwc-calendar-event:
We need to be able to omit the suffix of text that "overflows" from the event because it's too long. Is it possible to add to `vwc-calendar-event` a behavior similar to the `text-overflow: ellipsis` of CSS?

**Comment by yinonov** [comment]
@jshenkman how are colors applied to event card? should this be set by a form of connotation shades? should events expose color customization? and if so, should it also expose the "on-color" (you can't have 1 without the other)?
